[ingress-nginx] backport the nil://example.com redirect fix (#2970)
deckhouse/deckhouse: modules/402-ingress-nginx/images/controller-1-1/patches/README.md

https://github.com/kubernetes/ingress-nginx/pull/8213

### Nginx tmpl backport from 1.2

> Without this fix, redirects don't work if using behindL7Proxy controller and a load balancer without X-Forwarded-Proto header. They redirect to nil://example.com/abc.

Reviewer: ```suggestion
Without this fix, redirects don't work if using behindL7Proxy controller and a load balancer without X-Forwarded-Proto header. In this case, ingress-nginx redirects to nil://example.com/abc.
```
